store/ssmstore.go

```diff
@@ -241,6 +277,26 @@ func (s *SSMStore) WriteTags(ctx context.Context, id SecretId, tags map[string]s
 	return nil
 }
 
+func (s *SSMStore) checkForPresentRequiredTags(ctx context.Context, currentTags map[string]string, tags map[string]string) error {
+	requiredTags, err := requiredTags(ctx, s)
+	if err != nil {
+		return err
+	}
+	var missingTags []string
+	for _, requiredTag := range requiredTags {
+		_, alreadyPresent := currentTags[requiredTag]
+		_, beingUpdated := tags[requiredTag]
+		if alreadyPresent && !beingUpdated {
+			missingTags = append(missingTags, requiredTag)
+		}
+	}
```

Comment:
Would you mind elaborate a bit more on this for loop, especially on the if statement? I thought that if a required tag is already present, and not being updated, we should allow it given that the tags already exist.

Reply:
You're right, unless the option is selected to "delete other tags". When that happens, then any required tags which are already present must be explicitly provided again ... or else they would be deleted! This function `checkForPresentRequiredTags` is only called when "delete other tags" is true, so that's why this condition is OK.

This explanation isn't obvious from the code, though, so I'll add some comments to clarify.

Reply:
Thanks for clarifying it, Bill! I am wondering if we can not ask/require the user to provide the existing required tag when it already exists?
Or if this is only for terraform, i.e the tags will always exists when applying the changes?

Reply:
I don't think I understand your question. Let me lay out the way it works to see if I can answer it. Suppose that tag A is required and is already set, and someone runs `chamber tag write <svc> <key> B=valueb`.

* If they do not include `--delete-other-tags`, then the write should succeed. Tag A stays with its current value, and tag B is set to valueb.
* If they do include `--delete-other-tags`, then the write should fail. Chamber will say that tag A is required, so you have to provide a value for it (rewrite it).
* If they do include `--delete-other-tags` and also provide some value for tag A as well (`A=valueA B=valueB`), then the write should succeed. Tags A and B will both be set. So I think that covers requiring the user to provide the existing required tag.

Yeah, I guess Terraform is a good use case for `--delete-other-tags` because then chamber ensures that only exactly the provided tags exist, and all others would be deleted.